packages/astro/src/core/routing/astro-designed-error-pages.ts

```diff
@@ -0,0 +1,19 @@
+import type { ManifestData } from "../../@types/astro.js";
+
+export function ensure404Route(manifest: ManifestData) {
+	if (!manifest.routes.some(route => route.route === '/404')) {
+		manifest.routes.push({
+			component: 'astro-default-404',
```

Comment:
Let's export this `'astro-default-404'` as a constant, so we use this constant in all the files that check for this string.